iogt/views.py

```diff
@@ -108,3 +110,99 @@ def get(self, request):
             tuple(jsi18n_urls)
         )
         return Response(sitemap)
+
+
+class PageTreeAPIView(APIView):
+    def _get_renditions(self, image_id):
+        image_urls = []
+        for rendition in Rendition.objects.filter(image_id=image_id):
+            image_urls.append(f'{settings.MEDIA_URL}{rendition.file.name}')
+        return image_urls
+
+    def _get_banner_image_urls(self, page):
+        image_urls = []
+        if page.banner_image:
+            image_urls += self._get_renditions(page.banner_image)
+        return image_urls
+
+    def _get_lead_image_urls(self, page):
+        image_urls = []
+        if page.lead_image:
+            image_urls += self._get_renditions(page.lead_image)
+        return image_urls
+
+    def _get_image_icon_urls(self, page):
+        image_urls = []
+        if page.image_icon:
+            image_urls += self._get_renditions(page.image_icon)
+        return image_urls
+
+    def _get_stream_data_image_urls(self, stream_data):
+        image_urls = []
+        for block in stream_data:
+            if block['type'] == 'image':
+                image_urls += self._get_renditions(block['value'])
+            if block['type'] == 'paragraph':
+                tags = BeautifulSoup(block['value'], "html.parser").find_all('embed')
+                for tag in tags:
+                    if tag.attrs['embedtype'] == 'image':
+                        image_urls += self._get_renditions(tag.attrs['id'])
+        return image_urls
+
+    def _get_page_image_urls(self, page):
+        from home.models import BannerPage, Section, Article, OfflineAppPage
+
+        image_urls = []
+        if isinstance(page, (BannerPage,)):
+            image_urls += self._get_banner_image_urls(page)
+        if isinstance(page, (Section, Article, OfflineAppPage)):
+            image_urls += self._get_lead_image_urls(page)
+        if isinstance(page, (Section, Article, OfflineAppPage, Poll, Survey, Quiz)):
+            image_urls += self._get_image_icon_urls(page)
+        if isinstance(page, (Article, OfflineAppPage)):
+            image_urls += self._get_stream_data_image_urls(page.body.stream_data)
+        if isinstance(page, (Poll, Survey, Quiz)):
+            image_urls += self._get_stream_data_image_urls(page.description.stream_data)
+            image_urls += self._get_stream_data_image_urls(page.thank_you_text.stream_data)
+        return image_urls
+
+    def get(self, request, page_id):
+        from home.models import HomePage, Section, Article, OfflineAppPage, SVGToPNGMap
```

Comment:
is there a good reason why the import is here and not at the top of the file?

Reply:
this is due to circular import

Reply:
ok,

- views.py is loaded
- views.py loads models.py (from models import HomePage, ...)
- models.py loads views.py (from views import check_user_session)
- hence circular dependency

but I think the code (check_user_session) that is causing this is obsolete now
also, it should have been in the utils.py file not in view.

check_user_session: it was used to show a welcome banner to first-time users but I think it's not relevant anymore but Chris can confirm this.

Reply:
done